common/app/model/EmailAddons.scala

```diff
@@ -2,108 +2,122 @@ package model
 
 import conf.Static
 
-sealed trait EmailContent extends Product with Serializable {
+
+sealed trait EmailMetadata[T] extends Product with Serializable {
   def name: String
-  def banner: String
-  def test(c: ContentType): Boolean
+  def banner: Option[String]
```

Comment:
v minor but you could provide a default of `None` here, as with `address`, to avoid having to specify it in the implementation